app/src/ingestion/get_grouped_texts.py

```diff
@@ -0,0 +1,35 @@
+import logging
+
+from src.ingestion.pdf_elements import EnrichedText, TextType
+
+logger = logging.getLogger(__name__)
+
+
+def get_grouped_texts(markdown_texts: list[EnrichedText]) -> list[EnrichedText]:
+    """
+    Given EnrichedTexts, concatenate list tems together
+    with each other and with the preceeding NarrativeText.
+    """
+
+    if not markdown_texts:
+        return []
+
+    grouped_texts = [markdown_texts[0]]
+
+    for current_text in markdown_texts[1:]:
+        previous_text = grouped_texts[-1]
+
+        if current_text.type == TextType.LIST_ITEM:
```

Comment:
Need to check if `previous_text.text.rstrip()` ends in `:`.

Requirement 4 states:
> For list items: Keep grouped list items, nested list items, + the preceding sentence (which should end in a : ) together

Also
> [TASK 3.a] Split overly long text in grouped_texts (i.e., which don’t fit within the embedding model limit) into splits (a list of strings)
> - If a list is split, prepend the list-preceding sentence (ending with :) to the beginning of each splits[i].